plugin/evm/block.go

```diff
@@ -381,10 +382,7 @@ func (b *Block) verifyPredicates(predicateContext *precompileconfig.PredicateCon
 		return fmt.Errorf("failed to marshal predicate results: %w", err)
 	}
 	extraData := b.ethBlock.Extra()
-	headerPredicateResultsBytes, ok := predicate.GetPredicateResultBytes(extraData)
-	if !ok {
-		return fmt.Errorf("failed to find predicate results in extra data: %x", extraData)
-	}
```

Comment:
`predicateResultsBytes` is guaranteed not to be empty here, so this check is not required.

Reply:
Yes, exactly